src/filter/opt.rs

```diff
@@ -276,6 +276,22 @@ fn iterate(filter: Filter) -> Filter {
     filter
 }
 
+fn is_prefix(op: Op) -> bool {
+    match op {
+        Op::Prefix(_) => true,
+        _ => false,
+    }
+}
+
+fn prefix_of(op: Op) -> Filter {
```

Comment:
I'm not sure about the name here... This gets the last of a chain, if it is a prefix?

Reply:
maybe last_prefix?